### PR TITLE
upgrade to natchez 0.3.x

### PR DESCRIPTION
**Pull request by massimosiani**
Hi!
I'm trying to bump natchez to the latest version.
I'm unsure whether there are binary incompatibilities here.
Also, I'm not an expert in Datadog tracing, so I'd need a double check on that part.

### CODE REVIEW

**Comment by barryoneill** [comment]
Hi - it would be great to see this make it through, it's starting to age out. :)

**Comment by barryoneill** [comment]
@voidcontext is there any chance of publishing this, even as a milestone release?   Otherwise I'm going to have to fork this and publish internally which makes me feel bad :)

**Comment by voidcontext** [comment]
@barryoneill @massimosiani `8.0.0-M1` has been tagged and released, it should be synced to maven central shortly.

**Comment by barryoneill** [comment]
@voidcontext very much appreciated, thank you!

**Comment by voidcontext** [comment]
HI @massimosiani , first of all thank you for keeping this PR up to date!

We allocated some time in the coming weeks, to get this PR reviewed and merged. Meanwhile it seems there are some merge conflicts, if you could have a look at those that'd be cool.

**Comment by massimosiani** [comment]
Done! Thanks for your efforts on the project!

**Review by voidcontext — COMMENTED** [review]
Hi @massimosiani, we started the review, there are some initial comments. We are planning to continue later this week.

File: natchez-extras-slf4j/src/main/scala/com/ovoenergy/natchez/extras/slf4j/Slf4jSpan.scala [hunk 1]
```diff
@@ -52,7 +61,7 @@ object Slf4jSpan {
     Sync[F]
       .fromEither(
         k.toHeaders
-          .find(_._1.toLowerCase == "x-trace-token")
+          .find(_._1.toString.toLowerCase == "x-trace-token")
```
Comment (voidcontext) [inline natchez-extras-slf4j/src/main/scala/com/ovoenergy/natchez/extras/slf4j/Slf4jSpan.scala:64]: This should be simplified to:

```scala
          .find(_._1 == ci"x-trace-token")
```

Comment (voidcontext) [inline natchez-extras-slf4j/src/main/scala/com/ovoenergy/natchez/extras/slf4j/Slf4jSpan.scala:64]: This should be simplified to

```scala
          .find(_._1 == ci"x-trace-token")
```

File: natchez-extras-testkit/src/main/scala/com/ovoenergy/natchez/extras/testkit/TestEntryPoint.scala [hunk 2]
```diff
@@ -46,6 +47,11 @@ object TestEntryPoint {
               def spanId: F[Option[String]] = F.pure(None)
               def traceUri: F[Option[URI]] = F.pure(None)
               def kernel: F[Kernel] = F.pure(kern)
+              override def log(fields: (String, TraceValue)*): F[Unit] = put(fields: _*)
```
Comment (voidcontext) [inline natchez-extras-testkit/src/main/scala/com/ovoenergy/natchez/extras/testkit/TestEntryPoint.scala:50]: It's OK to add the `override` keyword as this is something we might want to do in the future, but we need to be consistent, so either add it everywhere in this file, or we don't add it at all.

**Review by voidcontext — COMMENTED** [review]
@massimosiani We added a few more comments, other than those we realised the use of `Span.Options` is missing in some cases, but that's not blocking this PR, we can make use of those in follow up changes.

Once those are fixed, we're happy to approve this PR, thanks for your contribution!

**Pull request merged**